### DISCUSSION TITLE
Delete `control-plane, master` ROLE node still able to query.

### DISCUSSION THREAD

**Discussion by BhautikChudasama** [k3s-io/k3s#7053] [Q&A]
### Describe Issue
I have k3s installed with `3` nodes. One is a master, and the other two are workers. I delete the master node still I am able to do queries. From my understanding, my expected behavior is I should not able to do a query. Please correct me If I am wrong If I delete the master nodes other components such as the scheduler, API server, and controller run as it is on the master node?

I also saw journalctl logs nothing suspicious.

### Other Information
OS: ubuntu 22.04.01
version: v1.25.6+k3s1

**Comment by brandond** [comment]
>  I delete the master node still I am able to do queries

Just deleting the node from the cluster doesn't actually stop the apiserver. Everything probably won't function correctly until you restart K3s and the node is recreated, but the apiserver itself doesn't rely on having a node object defined for the host that it's running on. You can actually start k3s with the --disable-agent flag and there will not be a node for the server at all.

**Reply by BhautikChudasama** [reply]
I can query the get pods, and server respond. Does it means deleting master node API server, scheduler, and running as it is on master!
